homeassistant/components/homekit/type_sensors.py

```diff
@@ -166,6 +180,58 @@ def async_update_state(self, new_state):
                 _LOGGER.debug("%s: Set air_quality to %d", self.entity_id, air_quality)
 
 
+@TYPES.register("PM10Sensor")
+class PM10Sensor(AirQualitySensor):
+    """Generate a PM10Sensor accessory as PM 10 sensor."""
+
+    def init(self):
+        """Override the init function for PM 10 Sensor."""
+        serv_air_quality = self.add_preload_service(
+            SERV_AIR_QUALITY_SENSOR, [CHAR_PM10_DENSITY]
+        )
+        self.char_quality = serv_air_quality.configure_char(CHAR_AIR_QUALITY, value=0)
+        self.char_density = serv_air_quality.configure_char(CHAR_PM10_DENSITY, value=0)
+
+    @callback
+    def async_update_state(self, new_state):
+        """Update accessory after state change."""
+        density = convert_to_float(new_state.state)
+        if density:
```

Comment:
```suggestion
        if not density:
            return
```

Then outdent the block below